### ISSUE TITLE
Ninja build broken when using commands with $-escaped characters

### ISSUE THREAD

**Issue by claudiofantacci** [robotology/ycm#240]
`ninja` build is broken with following error:
`ninja: error: build.ninja:244: bad $-escape (literal $ must be written as $$)`

The problem is the `--switch $(COLOR)` generated by CMake for the custom command in `AddUninstallTarget` and `YCMEPHelper`. Here is the list of such commands:
```bash
❯ git grep "(COLOR" ..
../modules/AddUninstallTarget.cmake:                  COMMAND "${CMAKE_COMMAND}" -E cmake_echo_color --switch=$\(COLOR\) --cyan "Uninstall the project..."
../modules/YCMEPHelper.cmake:                             COMMAND ${CMAKE_COMMAND} -E cmake_echo_color --switch=$(COLOR) --cyan "Working directory: ${source_dir}"
../modules/YCMEPHelper.cmake:    set(_source_cmd COMMAND ${CMAKE_COMMAND} -E cmake_echo_color --switch=$(COLOR) --cyan "${_name} SOURCE directory: "
../modules/YCMEPHelper.cmake:    set(_source_cmd COMMAND ${CMAKE_COMMAND} -E cmake_echo_color --switch=$(COLOR) --cyan "${_name} REPOSITORY directory: "
../modules/YCMEPHelper.cmake:                    COMMAND ${CMAKE_COMMAND} -E cmake_echo_color --switch=$(COLOR) --cyan "${_name} SOURCE directory: "
../modules/YCMEPHelper.cmake:  set(_binary_cmd COMMAND ${CMAKE_COMMAND} -E cmake_echo_color --switch=$(COLOR) --cyan "${_name} BINARY directory: "
```

Possible solutions:
 1. Remove them
 2. use `$$\(COLOR\)`
 3. use `$\{COLOR\}`

Related issue https://github.com/robotology/robotology-superbuild/issues/83

**Comment by claudiofantacci** [comment]
cc @diegoferigo

**Comment by drdanz** [comment]
Should be fixed in f2c61b1b9f5b2a7935bf79d1d43a68a218225804
